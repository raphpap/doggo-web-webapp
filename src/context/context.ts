// Vendor
import React from 'react';

// Types
import {ApplicationContext, Card} from './types';

const defaultValue: ApplicationContext = {
  actions: {
    battle: (ownCard: Card, opponentCard: Card) => Promise.reject(),
    capture: (name: string, image: string) => Promise.reject(),
    login: (username: string, password: string) => Promise.reject(),
  },
  state: {
    cards: null,
    error: null,
    loading: false,
    opponent: null
  }
};

const Context = React.createContext(defaultValue);

export const Provider = Context.Provider;
export const Consumer = Context.Consumer;
