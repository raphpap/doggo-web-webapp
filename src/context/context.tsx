// Vendor
import React from 'react';

// Types
import {
  ApplicationContext,
  BattlePayload,
  BattleStatus,
  CapturePayload,
  Card,
  GetNextOpponentPayload,
  LoginPayload
} from './types';

const defaultValue: ApplicationContext = {
  actions: {
    battle: (payload: BattlePayload) => Promise.reject(),
    capture: (payload: CapturePayload) => Promise.reject(),
    getNextOpponent: (payload: GetNextOpponentPayload) => Promise.reject(),
    login: (payload: LoginPayload) => Promise.reject(),
    selectBattleCard: (card: Card) => Promise.reject(),
    unselectBattleCard: () => Promise.reject()
  },
  state: {
    battle: {
      cardId: null,
      opponent: null,
      status: BattleStatus.NoOpponent
    },
    cards: null,
    error: null,
    loading: false
  }
};

const Context = React.createContext(defaultValue);

export const Provider = Context.Provider;
export const Consumer = Context.Consumer;
