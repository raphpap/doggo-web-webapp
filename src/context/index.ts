export * from './application-context-provider';
export * from './context';
export * from './types';
export * from './with-application-context';
