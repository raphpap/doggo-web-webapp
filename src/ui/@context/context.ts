// Vendor
import React from 'react';

// Types
import {ApplicationContext} from './types';

const defaultValue: ApplicationContext = {
  actions: {
    capture: (card: {name: string; image: string}) => Promise.reject(),
    login: (username: string, password: string) => Promise.reject()
  },
  state: {
    cards: null,
    error: null,
    loading: false
  }
};

const Context = React.createContext(defaultValue);

export const Provider = Context.Provider;
export const Consumer = Context.Consumer;
