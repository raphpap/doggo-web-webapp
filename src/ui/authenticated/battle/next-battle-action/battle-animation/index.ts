export * from './battle-animation';
export {default} from './battle-animation';
