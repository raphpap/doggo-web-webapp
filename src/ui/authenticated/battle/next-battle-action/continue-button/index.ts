export * from './continue-button';
export {default} from './continue-button';
