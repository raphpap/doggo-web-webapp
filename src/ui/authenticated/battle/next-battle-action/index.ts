export * from './next-battle-action';
export {default} from './next-battle-action';
