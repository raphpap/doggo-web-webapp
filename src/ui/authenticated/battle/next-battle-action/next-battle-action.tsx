// Vendor
import React from 'react';

// Assets
import battleLogo from 'doggo/assets/images/battle-logo.png';
import lostLogo from 'doggo/assets/images/lost-logo.png';
import wonLogo from 'doggo/assets/images/won-logo.png';

// SharedComponents
import Logo from 'doggo/ui/@components/logo';
import Message from 'doggo/ui/@components/message';
import PrettyButton from 'doggo/ui/@components/pretty-button';

// Components
import ActionContainer from './action-container';
import BattleAnimation from './battle-animation';
import ContinueButton from './continue-button';

// Context
import {BattleStatus, Card} from 'doggo/context';

// Types
interface Props {
  battleStatus: BattleStatus;
  onBattleClicked: (card: Card) => void;
  onNextAfterLost: () => void;
  onNextAfterWin: () => void;
  ownCard: Card | null;
}

export class NextBattleAction extends React.Component<Props> {
  public render() {
    const {
      battleStatus,
      onBattleClicked,
      onNextAfterLost,
      onNextAfterWin,
      ownCard
    } = this.props;

    return (
      <ActionContainer>
        {battleStatus === BattleStatus.NoOpponent && (
          <Message>Loading...</Message>
        )}

        {battleStatus === BattleStatus.NoCardSelected && (
          <Message>Get ready to battle</Message>
        )}

        {battleStatus === BattleStatus.Ready && (
          <PrettyButton
            logo={battleLogo}
            onClick={() => onBattleClicked(ownCard!)}
          />
        )}

        {battleStatus === BattleStatus.Ongoing && <BattleAnimation />}

        {battleStatus === BattleStatus.Won && (
          <div>
            <Logo logo={wonLogo} height={'110px'} marginUnits={0.1} />
            <ContinueButton onClick={onNextAfterWin}>
              {'Press to continue >>'}
            </ContinueButton>
          </div>
        )}

        {battleStatus === BattleStatus.Lost && (
          <div>
            <Logo logo={lostLogo} height={'110px'} marginUnits={0.1} />
            <ContinueButton onClick={onNextAfterLost}>
              {'Press to continue >>'}
            </ContinueButton>
          </div>
        )}
      </ActionContainer>
    );
  }
}

export default NextBattleAction;
