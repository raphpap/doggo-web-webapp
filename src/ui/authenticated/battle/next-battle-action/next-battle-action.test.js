// Vendor
import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

// Context
import {BattleStatus} from 'doggo/context';

// Components
import {NextBattleAction} from './next-battle-action';

// Mocks
import {mockCard} from 'doggo/test-mocks';

describe('<NextBattleAction />', () => {
  let wrapper;
  let battleStatus;
  let onBattleClicked;
  let onNextAfterLost;
  let onNextAfterWin;
  let ownCard;

  let props;

  beforeEach(() => {
    battleStatus = BattleStatus.NoOpponent;
    onBattleClicked = jest.fn();
    onNextAfterLost = jest.fn();
    onNextAfterWin = jest.fn();
    ownCard = null;

    props = {
      battleStatus,
      onBattleClicked,
      onNextAfterLost,
      onNextAfterWin,
      ownCard
    };

    wrapper = shallow(<NextBattleAction {...props} />);
  });

  it('renders correctly when no opponent', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly when no card selected', () => {
    battleStatus = BattleStatus.NoCardSelected;
    wrapper.setProps({battleStatus});
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly when ready to battle', () => {
    battleStatus = BattleStatus.Ready;
    wrapper.setProps({battleStatus});
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly when ongoing battle', () => {
    battleStatus = BattleStatus.Ongoing;
    wrapper.setProps({battleStatus});
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly when won battle', () => {
    battleStatus = BattleStatus.Won;
    wrapper.setProps({battleStatus});
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly when lost battle', () => {
    battleStatus = BattleStatus.Lost;
    wrapper.setProps({battleStatus});
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
