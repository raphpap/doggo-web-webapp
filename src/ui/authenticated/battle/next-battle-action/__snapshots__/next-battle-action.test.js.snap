// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NextBattleAction /> renders correctly when lost battle 1`] = `
<ActionContainer>
  <div>
    <Logo
      height="110px"
      logo="lost-logo.png"
      marginUnits={0.1}
    />
    <ContinueButton
      onClick={[MockFunction]}
    >
      Press to continue &gt;&gt;
    </ContinueButton>
  </div>
</ActionContainer>
`;

exports[`<NextBattleAction /> renders correctly when no card selected 1`] = `
<ActionContainer>
  <Message>
    Get ready to battle
  </Message>
</ActionContainer>
`;

exports[`<NextBattleAction /> renders correctly when no opponent 1`] = `
<ActionContainer>
  <Message>
    Loading...
  </Message>
</ActionContainer>
`;

exports[`<NextBattleAction /> renders correctly when ongoing battle 1`] = `
<ActionContainer>
  <BattleAnimation />
</ActionContainer>
`;

exports[`<NextBattleAction /> renders correctly when ready to battle 1`] = `
<ActionContainer>
  <PrettyButton
    logo="battle-logo.png"
    onClick={[Function]}
  />
</ActionContainer>
`;

exports[`<NextBattleAction /> renders correctly when won battle 1`] = `
<ActionContainer>
  <div>
    <Logo
      height="110px"
      logo="won-logo.png"
      marginUnits={0.1}
    />
    <ContinueButton
      onClick={[MockFunction]}
    >
      Press to continue &gt;&gt;
    </ContinueButton>
  </div>
</ActionContainer>
`;
